auction-app-backend/src/main/java/com/example/auctionapp/service/implementation/ProductServiceImpl.java

```diff
@@ -59,27 +57,36 @@ public ProductServiceImpl(final ProductRepository productRepository,
     }
 
     @Override
-    public ProductSearchResponse getProducts(UUID categoryId, String searchProduct, int page, int size) {
-        Pageable pageable = PageRequest.of(page, size);
-        Specification<ProductEntity> specification = ProductSpecification.withDynamicQuery(categoryId, searchProduct);
+    public ProductSearchResponse getProducts(final GetProductRequest getProductRequest) {
+        Sort sort = null;
+
+        if (getProductRequest.getSortField() != null && getProductRequest.getSortDirection() != null) {
```

Comment:
Let's move sort also to the util class, so that we avoid doing it here, in service we should only do this 
`final Page<Product> products = productRepository.findAll(ProductSpecification.buildSpecification(getProductRequest), pageable)
                                            .map(ProductEntity::toDomainModel);`
                                           Rest of this should be done in the util class